### ISSUE TITLE
convert pixmap to QImage without alpha

### ISSUE THREAD

**Issue by cges30901** [rk700/PyMuPDF#244]
Hello, I am working on a PyQt5 program. I am using python-poppler-qt5 to render PDF and display it, but I am considering switching to PyMuPDF. I can successfully convert PyMuPDF pixmap to QImage without setting `alpha = False`. However, the documentation say that disabling alpha saves memory and time. If I disable alpha, I cannot get it to work. I tried many different [QImage formats](http://doc.qt.io/qt-5/qimage.html#Format-enum) (with alpha enabled, I use QImage.Format_ARGB32), but the QImage is not correct. Here is my code:

```
import sys
from PyQt5.QtWidgets import QApplication, QLabel
from PyQt5.QtGui import QImage, QPixmap
import fitz

app = QApplication(sys.argv)

#render pdf with PyMuPDF
document = fitz.open("test.pdf")
page = document.loadPage(0)
pix = page.getPixmap()
#This does not work
#pix = page.getPixmap(alpha = False)
samples = pix.samples

#convert to QImage
image=QImage(samples, pix.width, pix.height, QImage.Format_ARGB32);

#show image in QLabel
label=QLabel()
label.setPixmap(QPixmap.fromImage(image))
label.show()
    
sys.exit(app.exec_())
```

I am using PyMuPDF 1.14.4 with PyQt 5.11.3. Do you know how to do this? Thank you!

**Comment by JorjMcKie** [comment]
Hello,
the pixel format of MuPDF pixmaps **always** has the alpha channel **appended** to the color bytes if present. The ``Pixmap.samples`` attribute has `0xRRGGBBAA` per pixel for RGB colorspaces with alpha channel.
So, `QImage.Format_RGBA8888` should work correctly work in this case. The value you used (`Format_ARGB32`) gives awkward colors in the cases I tried.

The pixel format of RGB colorspaces **without** alpha is `0xRRGGBB`. Therefore, I expected `Format_RGB888` to work. But it didn't until I found a different overloaded `QImage` constructor, that also accepts the **length of one image line** in bytes: `pix.stride`. This is my successfully working code:

```python
import sys
from PyQt5.QtWidgets import QApplication, QLabel
from PyQt5.QtGui import QImage, QPixmap
import fitz

app = QApplication(sys.argv)

#render pdf with PyMuPDF
document = fitz.open("test.pdf")
page = document[0] # short for loadPage(0)
pix = page.getPixmap(alpha=False)
#convert to QImage
image=QImage(pix.samples,
             pix.width, pix.height,
             pix.stride, # length of one image line in bytes
             QImage.Format_RGB888)

#show image in QLabel
label=QLabel()
label.setPixmap(QPixmap.fromImage(image))
label.show()
    
sys.exit(app.exec_())
```

Good luck and thanks for submitting this!

**Comment by JorjMcKie** [comment]
I did a quick comparison reading and "pixmapping" all 1310 pages of the Adobe manual. The following are total times (i.e. to be divided by 1310 to get times per page):
* Alpha=False
    - getPixmap = 4,0 sec
    - QImage = 0.77 sec
    - QPixmap = 0.79 sec
* Alpha=True
    - getPixmap = 4.18 sec
    - QImage = 1.1 sec
    - QPixmap = 0.9 sec
 So overall times are
* Alpha=False: 5.57
* Alpha=True: 6,15
In other words,, the non-alpha version is 9% to 10% faster and uses (at least) 25% less memory.

**Comment by cges30901** [comment]
> The value you used (`Format_ARGB32`) gives awkward colors in the cases I tried

You're right! I didn't notice the color is wrong because pdf I tried only contains text.

> But it didn't until I found a different overloaded `QImage` constructor, that also accepts the **length of one image line** in bytes

Thank you very much! It works great!
